src-java/kilda-persistence-tinkerpop/src/main/java/org/openkilda/persistence/ferma/repositories/FermaFlowRepository.java

```diff
@@ -622,4 +683,41 @@ protected void doRemove(FlowFrame frame) {
     protected FlowData doDetach(Flow entity, FlowFrame frame) {
         return Flow.FlowCloner.INSTANCE.deepCopy(frame, entity);
     }
+
+    /**
+     * Find all flows with filtering by flow IDs. This method does it as a single operation for better performance.
+     * @param flowIds Flow IDs for filtering.
+     * @return an empty collection if flowIds is empty,
+     *     otherwise returns a collection of Flows with the provided IDs if they exist.
+     */
+    private Collection<Flow> findFlowsByFlowIds(Collection<String> flowIds) {
```

Comment:
This seems to me as a good candidate for another public method in the parent interface. But the next method is very specific to a particular task and probably not reusable elsewhere. I was wondering if I need to create a separate class or it's ok like it is now. What are opinions?

Reply:
I agree to include it in the interface if you think is useful.